### PR TITLE
Material3

### PR DESCRIPTION
**Pull request by Satsrag**
for #50 

Adopt to material3.

### CODE REVIEW

**Comment by Satsrag** [comment]
Please don't merge it right now. This is to facilitate viewing the changed code. There are still some todos and tests that need to be done.

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -39,7 +39,7 @@ android {
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "net.studymongolian.example"
-        minSdkVersion 16
+        minSdkVersion flutter.minSdkVersion
         targetSdkVersion 29
```
Comment (suragch) [inline example/android/app/build.gradle:43]: We could probably update the target SDK to the Flutter default as well.

**Review by suragch — COMMENTED** [review]
You've done a lot of work! I didn't notice any major problems so far but I'll test it more when you finish your TODOs.

**Comment by suragch** [comment]
@Satsrag When do you think this will be ready to merge? I'm ok with merging even before it is ready for publishing as long as it is in a stable state. The reason I am asking is that I was thinking of upgrading to Flutter 3.22 but I wanted to wait if that will cause breaking changes that would conflict with this PR.

**Comment by Satsrag** [comment]
> @Satsrag When do you think this will be ready to merge? I'm ok with merging even before it is ready for publishing as long as it is in a stable state. The reason I am asking is that I was thinking of upgrading to Flutter 3.22 but I wanted to wait if that will cause breaking changes that would conflict with this PR.

Almost this week. You go on to upgrade to Flutter 3.22 first. If this PR conflicts with the master branch, I will fix this PR before merging.

**Comment by amjil** [comment]
> @Satsrag When do you think this will be ready to merge? I'm ok with merging even before it is ready for publishing as long as it is in a stable state. The reason I am asking is that I was thinking of upgrading to Flutter 3.22 but I wanted to wait if that will cause breaking changes that would conflict with this PR.

Flutter 3.22 encountered problem, waiting your , thank you.

**Comment by suragch** [comment]
@Satsrag I pushed a commit on master to fix a breaking change with Flutter 3.22. Hopefully that won't making merging your changes too difficult.

**Comment by Satsrag** [comment]
@suragch coding is done. Now all that's left is testing. Should we add `Unit Test` like Flutter way or just test manually？

**Comment by suragch** [comment]
Unit tests are definitely better because manual testing is often incomplete and likely to miss details in the future. However, it's also a lot of work to write tests in the first place. I'll go ahead and manually test what I can now. Anytime you want to add widget tests would be great though.

**Comment by Satsrag** [comment]
> Unit tests are definitely better because manual testing is often incomplete and likely to miss details in the future. However, it's also a lot of work to write tests in the first place. I'll go ahead and manually test what I can now. Anytime you want to add widget tests would be great though.

Agreeing with you. Now We test manually first. Please, have a look at the comments I have changed, I am not good at English. Thank you!

**Pull request merged**